app/controllers/invoices_controller.rb

```diff
@@ -36,6 +37,11 @@ def update
   end
  end
 
+ def view
+  @invoice = Invoice.find_by(token: params[:token])
+  render :show 
+end
```

Comment:
This isn't viewable if you're not logged in. If you are logged in, it's the same as the show page.

What I'd recommend is to extract the view-only characteristics of the show page into a partial and then you can load that without any user verification.

Reply:
This was the intention of the code. i only wanted the signed in users be able to see the content.

Reply:
This is from the project doc, and up the top of this PR.

> to share an invoice with external users who haven't signed in.

There's a way to do this using a `before_action` and methods from the authenticate gem: https://stackoverflow.com/questions/45511368/devise-how-to-allow-unauthenticated-access-to-some-pages